### PR TITLE
optimize: uniform error creation

### PR DESCRIPTION
**Pull request by justlorain**
#### What type of PR is this?

optimize: A new optimization

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: use package common/errors method to create error
zh(optional): 统一使用 common/errors 包的方法来创建 error

#### Which issue(s) this PR fixes:

null

### CODE REVIEW

File: pkg/common/tracer/stats/event.go [hunk 1]
```diff
@@ -88,8 +89,8 @@ var (
 
 // errors
 var (
-	ErrNotAllowed = errors.New("event definition is not allowed after initialization")
-	ErrDuplicated = errors.New("event name is already defined")
+	errNotAllowed = errors.NewPublic("event definition is not allowed after initialization")
+	errDuplicated = errors.NewPublic("event name is already defined")
```
Comment (L2ncE) [inline pkg/common/tracer/stats/event.go:93]: Will there be users who have used this public variable in the project? Will changing to private variable cause a break change.

Comment (justlorain) [reply pkg/common/tracer/stats/event.go]: Change to public, thx!

File: internal/bytesconv/errors.go [hunk 2]
```diff
@@ -41,13 +41,13 @@
...
+import "github.com/cloudwego/hertz/pkg/common/errors"
 
 var (
-	errEmptyInt               = errors.New("empty integer")
-	errUnexpectedFirstChar    = errors.New("unexpected first char found. Expecting 0-9")
-	errUnexpectedTrailingChar = errors.New("unexpected trailing char found. Expecting 0-9")
-	errTooLongInt             = errors.New("too long int")
-	errEmptyHexNum            = errors.New("empty hex number")
-	errTooLargeHexNum         = errors.New("too large hex number")
+	errEmptyInt               = errors.NewPublic("empty integer")
```
Comment (welkeyever) [inline internal/bytesconv/errors.go:47]: No need to change this - errors defined here act as core error, which is for cross-package.

Comment (justlorain) [reply internal/bytesconv/errors.go]: done, thx

**Comment by welkeyever** [comment]
Any progress about the UT may be updated here🙏 @wzekin

**Review by welkeyever — DISMISSED** [review]
LGTM, thx~

**Comment by justlorain** [comment]
The conflict has been resolved :)

File: pkg/common/test/mock/network.go [hunk 3]
```diff
@@ -143,6 +142,7 @@ func (m *SlowReadConn) Peek(i int) ([]byte, error) {
 		time.Sleep(100 * time.Millisecond)
 	}
 	if err != nil || len(b) != i {
+		time.Sleep(m.readTimeout)
```
Comment (welkeyever) [inline pkg/common/test/mock/network.go:145]: What  is the purpose of add this line?

Comment (justlorain) [reply pkg/common/test/mock/network.go:145]: This is the code incorporated by the develop branch.

Comment (welkeyever) [reply pkg/common/test/mock/network.go:145]: Check the latest develop branch here: https://github.com/cloudwego/hertz/blob/develop/pkg/common/test/mock/network.go#L146
Doesn't look consistent.

Comment (justlorain) [reply pkg/common/test/mock/network.go]: Removed. Sorry for my mistake.

**Review by welkeyever — APPROVED** [review]
LGTM

**Review by li-jin-gou — APPROVED** [review]

**Review by FGYFFFF — APPROVED** [review]

**Pull request merged**